nixos/modules/system/boot/kernel.nix

```
@@ -46,6 +46,9 @@ in
           kernelPatches = (originalArgs.kernelPatches or []) ++ kernelPatches;
           features = lib.recursiveUpdate super.kernel.features features;
         });
+        # Allows kernel patches to refer to the kernel version without introducing
+        # an infinite recursion.
+        _unpatchedKernel = super.kernel;
```

Why the underscore?

Maybe if you make it more intentional, it might be easier to manage in the future, when other changes come along.

```suggestion
        kernelVersion = super.kernel.version;
```

---

The goal was to differentiate between this attribute and actual kernel modules but I see how that will be confusing in the future.